File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        //create hashmap
        Map <Integer,Integer> map=new HashMap<>();

        for(int i=0;i<nums.length;i++){

            //get the complement using the target value 
            int complement=target - nums[i];

            //search the hashmap for complement ,if found ,we got our pair
            if(map.containsKey(complement)){
                return new int[]{map.get(complement),i};
            }

            //put the element in hashmap for subsequent searches 
            map.put(nums[i],i);
        }
        return new int []{0,0};
    }
}

File: 1-two-sum/two-sum.py
#nikhil lohiya
class Solution:
    def twoSum(self, nums, target):
        # Create hashmap (dictionary in Python)
        num_map = {}

        for i in range(len(nums)):
            # Get the complement using the target value
            complement = target - nums[i]

            # Search the hashmap for complement, if found, we got our pair
            if complement in num_map:
                return [num_map[complement], i]

            # Put the element in hashmap for subsequent searches
            num_map[nums[i]] = i

        return [0, 0]

        